### PR TITLE
Fixbug: client connection reset by deadline

### PR DESCRIPTION
**Pull request by longjiquan**
Signed-off-by: dragondriver <jiquan.long@zilliz.com>
issue: #14402 
/kind improvement

### CODE REVIEW

File: internal/util/grpcclient/client.go [hunk 1]
```diff
@@ -183,8 +183,14 @@ func (c *ClientBase) callOnce(ctx context.Context, caller func(client interface{
...
-	if err2 == context.Canceled || err2 == context.DeadlineExceeded {
-		return nil, err2
+
+	// status.Error(codes.Canceled, context.Canceled.Error()
+	// if err2 == context.Canceled || err2 == context.DeadlineExceeded {
+	// 	return nil, err2
+	// }
+
+	if !funcutil.CheckCtxValid(ctx) {
+		return ret, err2
```
Comment (czs007) [inline internal/util/grpcclient/client.go:193]: what ret is in this case? nil?

Comment (DragonDriver) [reply internal/util/grpcclient/client.go:193]: ret doesn't matter here. we can also choose to `return nil, err2`.

Comment (czs007) [reply internal/util/grpcclient/client.go:193]: ok, please return nil, err2 to avoid confusion.

**Pull request merged**